packages/vue-app/template/App.js

```diff
@@ -100,6 +101,28 @@ export default {
         }
       }
     },
+    async refreshPageData(showProgressBar) {
+      const context = this.$options.context
+      const pages = getMatchedComponentsInstances(this.$route)
+
+      if (!pages.length) {
+        return
+      }
+      this.$loading && this.$loading.start()
+      const promises = pages.map(async (page) => {
+        if (page.$options.fetch) {
+          await page.$options.fetch(context)
+        }
+        if (page.$options.asyncData) {
+          const newData = await page.$options.asyncData(context)
+          for (const key in newData) {
+            Vue.set(page.$data, key, newData[key])
+          }
+        }
+      })
+      await Promise.all(promises)
+      this.$loading && this.$loading.finish()
```

Comment:
Isn't error handling required in case of rejection of above await to finish loading in error state?

Reply:
Updated :)